daemon-scheduler/pkg/deployment/deployment.go

```diff
@@ -151,40 +156,54 @@ func (d deployment) CreateSubDeployment(ctx context.Context, environmentName str
 			"There is no deployment for environment with name '%s' to create a sub-deployment", environmentName)
 	}
 
-	return d.startDeployment(ctx, env, deployment, instanceARNs)
+	return d.StartDeployment(ctx, env, deployment, instanceARNs)
 }
 
-//TODO: wrap in a transaction so the environment and the deployment do not get modified in between being retrieved and starting tasks
-func (d deployment) startDeployment(ctx context.Context, env *types.Environment, deployment *types.Deployment, instanceARNs []*string) (*types.Deployment, error) {
+// TODO: wrap in a transaction so the environment and the deployment do not get modified in between being retrieved and starting tasks
+func (d deployment) StartDeployment(ctx context.Context, env *types.Environment, deployment *types.Deployment, instanceARNs []*string) (*types.Deployment, error) {
 	resp, err := d.ecs.StartTask(env.Cluster, instanceARNs, deployment.ID, deployment.TaskDefinition)
 	if err != nil {
 		return nil, errors.Wrapf(
 			err, "Error starting tasks for deployment with ID '%s' in environment with name '%s'", deployment.ID)
 	}
 
-	// if deployment is already completed then we do not update
+	failures := resp.Failures
+	if deployment.FailedInstances != nil {
+		failures = append(failures, deployment.FailedInstances...)
+	}
+
+	// if deployment is already completed (in the case where a new instance joins
+	// and there are no in-progress deployments) then we do not update the deployment object
 	// TODO: Figure out how we want to track failures in sub-deployments
 	if deployment.Status == types.DeploymentCompleted {
 		return deployment, nil
 	}
 
-	failures := resp.Failures
-	if deployment.FailedInstances != nil {
-		failures = append(failures, deployment.FailedInstances...)
+	if deployment.Status == types.DeploymentPending {
+		err = deployment.UpdateDeploymentInProgress(len(instanceARNs), failures)
```

Comment:
Shouldn't this method be called `UpdateDeploymentToInProgress`? I read this as updating a deployment that was already in progress before I jumped into the method to see what it did.